### PR TITLE
OCD-3279: ONC-ACB editing Promoting Interoperability data is Questionable Activity

### PR DESCRIPTION
**Pull request by kekey1**

### CODE REVIEW

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java [hunk 1]
```diff
@@ -588,6 +620,28 @@ private boolean isListingRealWorldTestingEligible(Long listingId) {
         return rwtEligServiceFactory.getInstance().getRwtEligibilityYearForListing(listingId, LOGGER).getEligibilityYear().isPresent();
     }
 
+    private boolean hasPromotingInteroperabilityHistoryChanged(CertifiedProductSearchDetails existingListing,
+            CertifiedProductSearchDetails updatedListing) {
+        existingListing.getPromotingInteroperabilityUserHistory().sort(new PromotingInteroperabilityComparator());
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java]: Does this need a sort? If not, we could remove the comparator too, right?

Comment (kekey1) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/questionableactivity/ListingQuestionableActivityProvider.java]: RIght!

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/ResourcePermissions.java [hunk 2]
```diff
@@ -463,6 +463,19 @@ public boolean doesUserHaveRole(String authority) {
         return role.getAuthority().equalsIgnoreCase(authority);
     }
 
+    public boolean doeAuditUserHaveRole(String authority) {
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/ResourcePermissions.java]:
```suggestion
    public boolean doesAuditUserHaveRole(String authority) {
```

**Review by andlar — DISMISSED** [review]
Happy with this going forward to test

**Review by tmy1313 — APPROVED** [review]

**Review by andlar — APPROVED** [review]

**Pull request merged**